wsmaster/che-core-api-factory-gitlab/src/main/java/org/eclipse/che/api/factory/server/gitlab/GitlabUrl.java

```diff
@@ -185,15 +183,22 @@ public String location() {
    * @return location of specified file in a repository
    */
   public String rawFileLocation(String fileName) {
-    StringJoiner joiner = new StringJoiner("/").add(hostName).add(username).add(project);
-    if (repository != null) {
-      joiner.add(repository);
-    }
-    joiner.add("-").add("raw").add(firstNonNull(branch, "master"));
-    if (subfolder != null) {
-      joiner.add(subfolder);
+    StringJoiner joiner =
+        new StringJoiner("/")
+            .add(hostName)
+            .add("api/v4/projects")
+            .add(username + "%2f" + project)
+            .add("repository")
+            .add("files")
+            .add(fileName)
+            .add("raw");
+    String resultUrl = joiner.toString();
```

Comment:
I guess we can do directly `String resultsUrl = new StringJoiner(..)....toString()`, without the extra `joiner` variable